Lib/fontTools/subset/cff.py

```diff
@@ -352,28 +352,45 @@ def processSubr(self, index, subr):
 				if subr_hints.status == 0:
 					hints.last_hint = index
 				else:
-					hints.last_hint = index - 2 # Leave the subr call in
+					hints.last_hint = index - 2  # Leave the subr call in
 
 		elif subr_hints.status == 0:
 			hints.deletions.append(index)
 
 		hints.status = max(hints.status, subr_hints.status)
 
+class StopHintCountEvent(Exception):
+	pass
+
+
+stop_hintcount_ops = ("op_hstem", "op_vstem", "op_rmoveto", "op_hmoveto",
+						"op_vmoveto")
+
+
 class _DesubroutinizingT2Decompiler(psCharStrings.SimpleT2Decompiler):
 
 	def __init__(self, localSubrs, globalSubrs, private=None):
-		psCharStrings.SimpleT2Decompiler.__init__(self,
-							  localSubrs,
-							  globalSubrs, private)
+		psCharStrings.SimpleT2Decompiler.__init__(self, localSubrs, globalSubrs,
+												private)
 
 	def execute(self, charString):
+		self.need_hintcount = True  # until proven otherwise
```

Comment:
I suggest moving the initialization of `need_hintcount` into the `__init__()` method.

Reply:
The problem with that is that the class instance can be re-used for more than one charstring. Running it on one charstring will leave self.need_hintcount set to False, and it needs to be True at the start of execute(). I could set the value in init, and then extend the parent class's reset() method to set this to True. However, since this class is pretty specialized, I chose to set close to were it is used.

Reply:
OK.